### PR TITLE
Lazily initialize/serialize collection properties

### LINKED ISSUE #7764: Consider returning non-null collections from KeyVault properties
The property below is null by default. Is that OK?

_Originally posted by @KrzysztofCwalina in https://github.com/Azure/azure-sdk-for-net/pull/7751_

### PR DESCRIPTION
**Pull request by heaths**
Fixes #7764

### CODE REVIEW

**Review by pakrym — CHANGES_REQUESTED** [review]
I don't think this change achieves the desired optimization. It replaces list and dictionary allocations with Lazy allocation. 

List and dictionary do not initialize the internal buffers in default ctors:

https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Collections/Generic/List.cs#L40
https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Collections/Generic/Dictionary.cs#L944

https://docs.microsoft.com/en-us/dotnet/api/system.threading.lazyinitializer.ensureinitialized?view=netframework-4.8 might be a more appropriate primitive to use here.

File: sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Azure.Core.Testing
+{
+    public class JsonStream : IDisposable
+    {
+        private readonly MemoryStream _buffer = new MemoryStream();
+
+        public string Content => Encoding.UTF8.GetString(_buffer.GetBuffer(), 0, (int)_buffer.Length);
+
+        public Utf8JsonReader CreateReader(JsonReaderOptions options = default) => new Utf8JsonReader(_buffer.GetBuffer(), options);
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs]: This method is never used.

File: sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.Core.Testing
+{
+    public class JsonStream : IDisposable
```
Comment (pakrym) [inline sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs]: Are we sure we need this type in shared code?

Comment (heaths) [reply sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs]: I had planned to beef it up eventually, but can move it out for now if you'd like. Figured it'd be a good helper class.

File: sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs (hunk 1 above)
Comment (heaths) [reply sdk/core/Azure.Core/tests/TestFramework/JsonStream.cs]: I had planned to beef it up eventually, but can move it out for now if you'd like. Figured it'd be a good helper class.

**Comment by heaths** [comment]
That's a good point. We used EnsureInitialized in the setup engine as well for low-overhead objects, which these are as you pointed out.

**Review by pakrym — APPROVED** [review]

**Pull request merged**